content/chainguard/chainguard-images/working-with-images/scanners/grype-tutorial/index.md

```diff
@@ -0,0 +1,250 @@
+
+```bash
+syft -o syft-json python > sbom.json
+cat sbom.json | grype               
+```
+
+## Comprehending Grype Output
+
+By default, grype output is divided into two sections, a summary of information on the scanned artifact and an itemized list of CVE.
+
+ In this section, we'll use an alpine version of the official Python image as an example. Since we're specifying a version, you may encounter more CVE when following the examples than are shown here, as CVE will accumulate over time.
+
+Scan the image with the following command:
+
+```bash
+grype python:3.10.14-alpine3.20
+```
+
+You should receive output similar to the following:
+
+```
+ ✔ Vulnerability DB                [no update av
+ ✔ Parsed image                    sha256:f48490
+ ✔ Cataloged contents              09feb83998b9d
+   ├── ✔ Packages                        [47 pac
+   │     └── ⠹ Linux kernel cataloger          
+   ├── ✔ File digests                    [659 fi
+   ├── ✔ File metadata                   [659 lo
+   └── ✔ Executables                     [144 ex
+ ✔ Scanned for vulnerabilities     [10 vulnerabi
+   ├── by severity: 0 critical, 1 high, 8 medium
+   └── by status:   7 fixed, 3 not-fixed, 0 igno
+NAME           INSTALLED   FIXED-IN    TYPE    VULNERABILITY        SEVERITY 
+busybox        1.36.1-r28  1.36.1-r29  apk     CVE-2023-42365       Medium    
+busybox        1.36.1-r28  1.36.1-r29  apk     CVE-2023-42364       Medium    
+busybox-binsh  1.36.1-r28  1.36.1-r29  apk     CVE-2023-42365       Medium    
+busybox-binsh  1.36.1-r28  1.36.1-r29  apk     CVE-2023-42364       Medium    
+pip            23.0.1      23.3        python  GHSA-mq26-g339-26xf  Medium    
+python         3.10.14                 binary  CVE-2023-36632       High      
+python         3.10.14                 binary  CVE-2023-27043       Medium    
+python         3.10.14                 binary  CVE-2024-4030        Unknown   
+ssl_client     1.36.1-r28  1.36.1-r29  apk     CVE-2023-42365       Medium    
+ssl_client     1.36.1-r28  1.36.1-r29  apk     CVE-2023-42364       Medium
+```
+
+### Interpreting the Summary
+
+In the initial portion of its results output, grype summarizes information on the scanned artifact and gives an overview of known vulnerabilities. In the case of a scanned image, the output includes the image digest, a unique hash of the image that can be used as an identifier. 
+
+Overview output includes the number of packages, files, and executables found in the artifact. Generally speaking, CVE are detected against packages, but the number of executables detected can also give you an idea of the attack surface of the scanned image or filesystem.
+
+Finally, this portion gives a count of the number of CVE detected by severity and fixed status. Severity categorization sorts CVE into four categories based on the Common Vulnerability Scoring System (CVSS).
+
+{{< details "What is CVSS??" >}}
```

Comment:
this was nifty!

Reply:
Been trying to do more blurbs since it saves effort later and I like the details so that folks who know the term can skip.